lib/absinthe/subscription.ex

```diff
@@ -175,10 +175,6 @@ defmodule Absinthe.Subscription do
   def unsubscribe(pubsub, doc_id) do
     registry = pubsub |> registry_name
 
-    for field_key <- pdict_fields(doc_id) do
-      Registry.unregister(registry, field_key)
-    end
```

Comment:
We still want to clean up the record in the registry, but this only removes the entry associated with the `doc_id` and not all subscriptions with the same `field_key`
```suggestion
    for field_key <- pdict_fields(doc_id) do
      Registry.unregister_match(registry, field_key, doc_id)
    end
```